scripts/resolve_container_path.py

```diff
@@ -19,24 +30,44 @@
 
 def resolve_container_path(container_path):
     container_path = os.path.realpath(container_path)
+    mountinfo = '/proc/self/mountinfo'
     try:
-        with open('/proc/self/cgroup') as f:
-            # Entries in /proc/self/cgroup look like this (note the nesting):
-            # 11:name=systemd:/docker/82c1bd2…23b5bcf/docker/6547bce…60ca5a7
-            prefix, container_id = next(f).strip().split(':')[2].split('/')[-2:]
+        with open(mountinfo) as f:
+            log.info('Found %s', mountinfo)
+            # Entries of interest in /proc/self/mountinfo look as follows:
+            # 752 744 259:2 /docker/containers/dc61d9…CONTAINER_ID…/<variable> …
+            prefix = '/docker/containers'
+            log.info('cgroup v2 prefix is %s', prefix)
+            containers = set()
+            for line in csv.reader(f, delimiter=' '):
```

Comment:
I'm pretty sure this file isn't actually meant to be CSV. With the CSV reader, you're pulling in a few more syntactic quirks like handling of double quotes around values that contain spaces, and line-ending oddities from 50 years ago. In this case, you should do the parsing by hand which means iterating over the lines (a file opened in text mode is an iterable of lines) and splitting each line on a literal space. Maybe you could add an assertion of the total number of items after the split.